packages/odyssey-react-mui/src/useAutocomplete.tsx

```diff
@@ -0,0 +1,398 @@
+        undefined,
+        IsCustomValueAllowed
+      >["onInputChange"]
+    >
+  >(
+    (event, value, reason) => {
+      onInputChangeProp?.(event, value, reason);
+    },
+    [onInputChangeProp],
+  );
+
+  const Row = ({
+    data,
+    index,
+    setItemSize,
+    style,
+  }: ListChildComponentProps & { setItemSize: SetItemSize }) => {
+    const rowRef = useRef<HTMLDivElement>(null);
+
+    useEffect(() => {
+      if (rowRef.current) {
+        /**
+         * Checking for child height to workaround a bug where the clientHeight of the row isn't updated correctly
+         * @see here if you need to know more: https://github.com/bvaughn/react-window/issues/582#issuecomment-1883074908
+         */
+        const firstChild = rowRef.current.firstElementChild;
+        const height = firstChild
+          ? firstChild.clientHeight
+          : rowRef.current.clientHeight;
+
+        setItemSize(height);
+      }
+    }, [index, rowRef, setItemSize]);
+
+    const baseOption = data[index];
+    const { key, props } = baseOption;
+
+    /**
+     * react-window calculates the absolute positions of the list items, via an inline style, so
+     * we need to add it to each list item that is being rendered in the viewable list window.
+     * @see here if you need to know more: https://github.com/bvaughn/react-window?tab=readme-ov-file#why-is-my-list-blank-when-i-scroll
+     */
+    const styles = useMemo(
+      () => ({
+        ...style,
+        height: "auto",
+      }),
+      [style],
+    );
+
+    return (
+      <div ref={rowRef}>
+        <li {...props} key={key} style={styles} />
+      </div>
+    );
+  };
+
+  const OuterListboxContext = createContext({});
+
+  const OuterListboxElementType = forwardRef<HTMLDivElement>((props, ref) => {
+    const outerProps = useContext(OuterListboxContext);
+    return <div ref={ref} {...props} {...outerProps} />;
+  });
+
+  const useResetCache = (length: number) => {
+    const resetCacheRef = useRef<VariableSizeList>(null);
+    useEffect(() => {
+      if (resetCacheRef.current) {
+        resetCacheRef.current.resetAfterIndex(0, true);
+      }
+    }, [length]);
+    return resetCacheRef;
+  };
+
+  const ListboxComponent = forwardRef<
+    HTMLDivElement,
+    HTMLAttributes<HTMLElement>
+  >(function (props, ref) {
+    const [listHeight, setListHeight] = useState(0);
+
+    const { children, ...other } = props;
+    const itemData: ReactElement[] = (children as ReactElement[]).flatMap(
+      (item: ReactElement & { children?: ReactElement[] }) =>
+        [item].concat(item.children || []),
+    );
+
+    const sizeMap = useRef<number[]>([]);
+
+    const getListBoxHeight = useCallback(() => {
+      // 8px of padding top/bottom applied by MUI
+      const COMBINED_LISTBOX_PADDING = 16;
+
+      if (itemData.length > OVERSCAN_ROW_COUNT) {
+        // has a max-height of 40vh set in CSS. This is only set because height needs to be a number
+        return 99999;
```

Comment:
Can we use `Infinity` instead of `99999`?

```suggestion
      if (itemData.length > OVERSCAN_ROW_COUNT) {
        // has a max-height of 40vh set in CSS. This is only set because height needs to be a number
        return Infinity;
```

Reply:
I can't use `Infinity` here. It's applying this number to a height under the hood and `Infinity` is invalid in that scenario